src/settings.lisp

```diff
@@ -0,0 +1,26 @@
+;;;; settings.lisp
+;;;;
+;;;; This file contains variables that can be used to control how
+;;;; Coalton works.
+
+(in-package #:coalton-impl)
+
+(deftype interaction-mode ()
+  `(member :development :release))
+(declaim (type interaction-mode *interaction-mode*))
+
+
+;; *interaction-mode* determines how redefinable code generated by Coalton is.
+;;
+;; In development mode Coalton turns all declare-type forms into CLOS
+;; objects to support a repl based workflow.
+;;
+;; In release mode Coalton compiles declare-type forms into frozen
+;; structs or even more optimal layouts which may not support
+;; redefinition.
+;;
+;; This variable *must* be set before the standard library is loaded.
```

Comment:
Are Coalton development mode and release mode mixable? Can one library be compiled as release and another as development? Can Coalton be compiled with release, and the rest of the world with development?

Reply:
In the current implementation I think that even within a single module different `coalton-toplevel` forms could be compiled in different modes. However I would prefer not to codify that as allowed behavior because we may want to do some form of whole program analysis in the future.